File: model/src/main/java/repository/AdminDataRepository.java
package repository;

import model.AdminData;

import java.time.YearMonth;
import java.util.Optional;
import java.util.concurrent.CompletionStage;

/**
 * Definition of an Admin Data Repository
 */
public interface AdminDataRepository {

    CompletionStage<YearMonth> getCurrentPeriod();

    CompletionStage<Optional<AdminData>> lookup(YearMonth referencePeriod, String key);

}

File: model/src/main/scala/repository/AdminDataRepository.scala
package repository

//import java.util.concurrent.CompletionStage

import scala.concurrent.Future

import com.github.nscala_time.time.Imports.YearMonth

import model.AdminData

/**
 * AdminDataRepository
 * ----------------
 * Author: haqa
 * Date: 23 October 2017 - 20:41
 * Copyright (c) 2017  Office for National Statistics
 */

trait AdminDataRepository {

  def getCurrentPeriod: Future[YearMonth]
  //  def getCurrentPeriod: CompletionStage[Option[YearMonth]]

  def lookup(referencePeriod: YearMonth, key: String): Future[Option[AdminData]]
  //  def lookup(referencePeriod: YearMonth, key: String): CompletionStage[AdminData]

}
